### PR TITLE
Remove more deprecated functionality

### PR DESCRIPTION
**Pull request by alexcrichton**
This removes a large amount of deprecated functionality (pre-1.0 beta). Mostly this cleans up `std::num`.

### CODE REVIEW

**Comment by brson** [comment]
I tagged this with 'beta' so we don't forget this needs to go in.

File: src/libcoretest/num/mod.rs [hunk 1]
```diff
@@ -56,33 +49,32 @@ mod test {
...
     #[test]
     fn from_str_issue7588() {
-        let u : Option<u8> = from_str_radix("1000", 10).ok();
+        let u : Option<u8> = u8::from_str_radix("1000", 10).ok();
         assert_eq!(u, None);
-        let s : Option<i16> = from_str_radix("80000", 10).ok();
+        let s : Option<i16> = i16::from_str_radix("80000", 10).ok();
         assert_eq!(s, None);
-        let f : Option<f32> = from_str_radix("10000000000000000000000000000000000000000", 10).ok();
+        let f : Option<f32> = f32::from_str_radix("10000000000000000000000000000000000000000", 10).ok();
```
Comment (tamird) [inline src/libcoretest/num/mod.rs]: this failed tidy

**Comment by pnkfelix** [comment]
going from nominated to (nominated, accepted)

**Pull request merged**